### PR TITLE
 Install and manage st2chatops

### PR DESCRIPTION
**Pull request by ruriky**
- Install needed packages
- Manage st2chatops.env file (hubot and adapter configuration)
- Install optional nodejs modules for hubot adapter
- Manage st2chatops service

Example hiera config for setting up hubot and rocketchat adapter:
```
st2::chatops_npm_packages:
  hubot-adapter:
    package: 'hubot-rocketchat'
    source: 'git+ssh://git@git.company.com:npm/hubot-rocketchat#master'
st2::chatops_adapter_config:
  HUBOT_ADAPTER: rocketchat
  ROCKETCHAT_URL: "https://test-chat.company.com:443"
  ROCKETCHAT_ROOM: 'sandbox'
  LISTEN_ON_ALL_PUBLIC: true
  ROCKETCHAT_USER: st2
  ROCKETCHAT_PASSWORD: xxxxxx
  ROCKETCHAT_AUTH: password
  RESPOND_TO_DM: true
```

I am successfully using this in my test env so it passes my use cases. Any comments are appreciated.

### CODE REVIEW

**Comment by ruriky** [comment]
Committed some style fixes to please Travis.

File: manifests/init.pp [hunk 1]
```diff
@@ -108,4 +108,6 @@
   $datastore_keys_dir       = $::st2::params::datstore_keys_dir,
   $datastore_key_path       = "${::st2::params::datstore_keys_dir}/datastore_key.json",
   $nginx_manage_repo        = true,
```
Comment (nmaludy) [inline manifests/init.pp]: Please add documentation to this class about the new parameters

File: manifests/profile/chatops.pp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+#  [*_chatops_env_file*] - Local scoped variable for full path of st2chatops env file.
+#
+# === Examples
+#
+#  include st2::profile::chatops
+#
+
+class st2::profile::chatops (
+  $version                      = $::st2::version,
+  $hubot_log_level              = 'debug',
```
Comment (nmaludy) [inline manifests/profile/chatops.pp]: Please place these defaults in ::st2::params, with an override in ::st2

File: manifests/profile/chatops.pp [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  include ::st2::profile::nodejs
+
+  $npm_package_defaults = {
+    ensure  => present,
+    target  => $_chatops_dir,
+    require => Class['St2::Profile::Nodejs'],
+    tag     => 'st2::chatops::npm_package',
+  }
+
+  create_resources('nodejs::npm', $npm_packages, $npm_package_defaults)
```
Comment (nmaludy) [inline manifests/profile/chatops.pp]: Please prefix classes with leading `::`, example: `::nodejs::npm`

File: manifests/profile/chatops.pp [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+#
+# === Parameters
+#
+#  [*version*]                      - Version of StackStorm to install
+#  [*hubot_log_level*]              - Hubot log level
+#  [*hubot_express_port*]           - Express port hubot listens to
+#  [*tls_cert_reject_unauthorized*] - Set to 1 when using self signed certs
+#  [*hubot_name*]                   - Hubot's name
+#  [*npm_packages*]                 - Nodejs packages to be installed (hubot adapter)
+#  [*adapter_config*]               - Configuration for hubot adapter
```
Comment (nmaludy) [inline manifests/profile/chatops.pp]: Please document the type (hash) and an example config

**Comment by ruriky** [comment]
Thank you for the review comments! I will squash the commits when we agree that the changes are ready to be merged.

**Comment by bdandoy** [comment]
Any update on this?

**Comment by nmaludy** [comment]
@bdandoy still waiting on updates based on my review from @ruriky

File: manifests/init.pp (hunk 1 above)
Comment (ruriky) [reply manifests/init.pp]: Added comments about the new chatops variables.

**Comment by ruriky** [comment]
I have added the review changes I missed at the first time. Sorry about that. Is it ok now?

**Comment by nmaludy** [comment]
Thanks for the great work, again!

**Pull request merged**